There must be exactly one bean that matches the skip predicate

### Describe the bug


``` java
@Slf4j
@ApplicationScoped
public class TestJob {
    @Scheduled(every="10s", skipExecutionIf = TestPredicate.class)
    void test() {
        log.info("Scheduled job execute");
    }
}
```

```java
@Singleton
public class TestPredicate implements Scheduled.SkipPredicate {
    @Override
    public boolean test(ScheduledExecution execution) {
        return false;
    }
}
```
I am going to test the skip function of quarkus-scheduler, when I start Quarkus, the following error is printed

```log
2021-09-03 19:08:53.617[ERROR]-[main]-[i.q.d.d.IsolatedDevModeMain(-1)]: Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: javax.enterprise.inject.spi.DeploymentException: java.lang.IllegalStateException: There must be exactly one bean that matches the skip predicate: "cn.ecpark.service.sample.quarkus.TestPredicate" on: @Scheduled(every = "10s",skipExecutionIf = cn.ecpark.service.sample.quarkus.TestPredicate)
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1094)
	at io.quarkus.arc.processor.BeanProcessor.processValidationErrors(BeanProcessor.java:142)
	at io.quarkus.arc.deployment.ArcProcessor.generateResources(ArcProcessor.java:447)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.IllegalStateException: There must be exactly one bean that matches the skip predicate: "cn.ecpark.service.sample.quarkus.TestPredicate" on: @Scheduled(every = "10s",skipExecutionIf = cn.ecpark.service.sample.quarkus.TestPredicate)
	at io.quarkus.scheduler.deployment.SchedulerProcessor.validateScheduled(SchedulerProcessor.java:412)
	at io.quarkus.scheduler.deployment.SchedulerProcessor.validateScheduledBusinessMethods(SchedulerProcessor.java:189)
	... 11 more
```

### Expected behavior

The log prints "Scheduled job execute" every 10 seconds

### Actual behavior

exception log

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.22000.168]

### Output of `java -version`

java version "11.0.10" 2021-01-19 

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.1.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1

### Additional information

_No response_

--- a/extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java
+++ b/extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java
@@ -96,6 +96,7 @@ public class SchedulerProcessor {
     static final DotName SCHEDULED_NAME = DotName.createSimple(Scheduled.class.getName());
     static final DotName SCHEDULES_NAME = DotName.createSimple(Scheduled.Schedules.class.getName());
     static final DotName SKIP_NEVER_NAME = DotName.createSimple(Scheduled.Never.class.getName());
+    static final DotName SKIP_PREDICATE = DotName.createSimple(Scheduled.SkipPredicate.class.getName());
 
     static final Type SCHEDULED_EXECUTION_TYPE = Type.create(DotName.createSimple(ScheduledExecution.class.getName()),
             Kind.CLASS);
@@ -464,6 +465,11 @@ public class SchedulerProcessor {
         }
 
         return null;
+    }
+
+    @BuildStep
+    UnremovableBeanBuildItem unremoveableSkipPredicates() {
+        return new UnremovableBeanBuildItem(new UnremovableBeanBuildItem.BeanTypeExclusion(SKIP_PREDICATE));
     }
 
 }

--- a/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/ConditionalExecutionTest.java
+++ b/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/ConditionalExecutionTest.java
@@ -1,11 +1,13 @@
 package io.quarkus.scheduler.test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import javax.enterprise.event.Observes;
 import javax.inject.Singleton;
@@ -44,17 +46,25 @@ public class ConditionalExecutionTest {
             Thread.currentThread().interrupt();
             throw new IllegalStateException(e);
         }
+
+        assertTrue(OtherIsDisabled.TESTED.get());
+        assertEquals(0, Jobs.OTHER_COUNT.get());
     }
 
     static class Jobs {
 
         static final CountDownLatch COUNTER = new CountDownLatch(1);
+        static final AtomicInteger OTHER_COUNT = new AtomicInteger(0);
 
         @Scheduled(identity = "foo", every = "1s", skipExecutionIf = IsDisabled.class)
         void doSomething() throws InterruptedException {
             COUNTER.countDown();
         }
 
+        @Scheduled(identity = "other-foo", every = "1s", skipExecutionIf = OtherIsDisabled.class)
+        void doSomethingElse() throws InterruptedException {
+            OTHER_COUNT.incrementAndGet();
+        }
     }
 
     @Singleton
@@ -74,6 +84,19 @@ public class ConditionalExecutionTest {
 
                 SKIPPED_LATCH.countDown();
             }
+        }
+
+    }
+
+    @Singleton
+    public static class OtherIsDisabled implements Scheduled.SkipPredicate {
+
+        static final AtomicBoolean TESTED = new AtomicBoolean(false);
+
+        @Override
+        public boolean test(ScheduledExecution execution) {
+            TESTED.set(true);
+            return true;
         }
 
     }

